UPGRADE.md

```diff
@@ -1,5 +1,9 @@
 # Upgrade to 2.9
 
+## Deprecated `Statement::fetchColumn()` with an invalid index
+
+Calls to `Statement::fetchColumn()` with an invalid column index currently return `NULL`. In the future, such calls will result in a exception.
```

Comment:
Well this is not entirely true considering that errors will be triggered by PHP itself and we are propagating them already. I would even question if this is a deprecation at all as it is undocumented behaviour. Our API does not tell anything about invalid index and `null`. Also why would someone expect such behaviour when it is clearly a bug in the client code? Even if someone would use `null` to tell whether the index in question is out of range, you couldn't even safely do that because `null` is not distinguishable from a regular index `null`.
IMO we could simply implement exceptions and maybe rather document it as "minor BC break" (even though it still is nothing we ever intended to support).

Reply:
IMO better to not tease consumers with sneaky exceptions, unless it's a security issue or so.